pkgs/development/libraries/aravis/default.nix

```diff
@@ -1,85 +1,77 @@
+assert enableViewer -> wrapGAppsHook != null;
 
-    outputs = [ "bin" "dev" "out" "lib" ];
+stdenv.mkDerivation rec {
+  pname = "aravis";
+  version = "0.8.21";
 
-    nativeBuildInputs = [
-      autoreconfHook
-      pkg-config
-      intltool
-      gtk-doc
-    ] ++ lib.optional enableViewer wrapGAppsHook;
+  src = fetchFromGitHub {
+    owner = "AravisProject";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-z4fuo8tVyHf2Bw73ZfAEpAYmzbr9UIzEWPC5f95wnD8=";
+  };
 
-    buildInputs =
-      [ glib libxml2 ]
-      ++ lib.optional enableUsb libusb1
-      ++ lib.optional enablePacketSocket audit
-      ++ lib.optionals (enableViewer || enableGstPlugin) [ gstreamer gst-plugins-base gst-plugins-good gst-plugins-bad ]
-      ++ lib.optionals (enableViewer) [ libnotify gtk3 gnome.adwaita-icon-theme ];
+  outputs = [ "bin" "dev" "out" "lib" ];
 
-    preAutoreconf = "./autogen.sh";
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gi-docgen
+  ] ++ lib.optional enableViewer wrapGAppsHook;
 
-    configureFlags =
-      lib.optional enableUsb "--enable-usb"
-        ++ lib.optional enablePacketSocket "--enable-packet-socket"
-        ++ lib.optional enableViewer "--enable-viewer"
-        ++ lib.optional enableGstPlugin
-        (if gstreamerAtLeastVersion1 then "--enable-gst-plugin" else "--enable-gst-0.10-plugin")
-        ++ lib.optional enableCppTest "--enable-cpp-test"
-        ++ lib.optional enableFastHeartbeat "--enable-fast-heartbeat"
-        ++ lib.optional enableAsan "--enable-asan";
+  buildInputs =
+    [ glib libxml2 gobject-introspection ]
+    ++ lib.optional enableUsb libusb1
+    ++ lib.optionals (enableViewer || enableGstPlugin) (with gst_all_1; [ gstreamer gst-plugins-base gst-plugins-good gst-plugins-bad ])
+    ++ lib.optionals (enableViewer) [ gtk3 hicolor-icon-theme ];
 
-    postPatch = ''
-        ln -s ${gtk-doc}/share/gtk-doc/data/gtk-doc.make .
-      '';
+  mesonFlags = [
+  ] ++ lib.optional enableFastHeartbeat "-Dfast-heartbeat=enabled"
+  ++ lib.optional (!enableGstPlugin) "-Dgst-plugin=disabled"
+  ++ lib.optional (!enableViewer) "-Dviewer=disabled"
+  ++ lib.optional (!enableUsb) "-Dviewer=disabled"
+  ++ lib.optional (!enablePacketSocket) "-Dpacket-socket=disabled";
 
-    doCheck = true;
+  doCheck = true;
 
-    meta = {
-      description = "Library for video acquisition using GenICam cameras";
-      longDescription = ''
-        Implements the gigabit ethernet and USB3 protocols used by industrial cameras.
-      '';
-      homepage = "https://aravisproject.github.io/docs/aravis-0.5";
-      license = lib.licenses.lgpl2;
-      maintainers = [];
-      platforms = lib.platforms.unix;
-    };
-  }
+  meta = {
+    description = "Library for video acquisition using GenICam cameras";
+    longDescription = ''
+      Implements the gigabit ethernet and USB3 protocols used by industrial cameras.
+    '';
+    homepage = "https://aravisproject.github.io/docs/aravis-0.8";
```

Comment:
I think it makes more sense to just refer to the root url rather than to the docs for the `homepage`.

Reply:
Turns out the root URL refers to a mostly blank page with no useful links or information. This is the best link that's not the Github repository page.